### PR TITLE
Feature/model monitoring

### PR DESCRIPTION
**Pull request by ranjan-stha**
Added scripts to calculate embeddings of excerpts, performance metrics at tag-level, per-project level, feature drift calculations per project.

### CODE REVIEW

File: monitoring_scripts/featuredrift.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import numpy as np
+from tqdm import tqdm
+from typing import List
+
+from evidently.report import Report
+from evidently.metrics import DatasetDriftMetric, DataDriftTable
+
+class FeatureDrift:
+    """
+    Calculates the per project feature drift of excerpts
```
Comment (k9845) [inline monitoring_scripts/featuredrift.py]:
```suggestion
    Calculates per project feature drift of excerpts
```

File: monitoring_scripts/featuredrift.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        """
+        Reads the embeddings and gets the list
+        Input: Embedding Series
+        Output: Embedding List
+        """
+        return embeddings.apply(eval).apply(np.array).to_list()
+    
+    def _project_id_based_mask(self,
+        df: pd.DataFrame,
+        project_id: float=None
```
Comment (k9845) [inline monitoring_scripts/featuredrift.py]: Why do we have `project_id` of type float?

File: monitoring_scripts/featuredrift.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                reference_df = self._project_id_based_mask(self.reference_df, project_id)
+                current_df = self._project_id_based_mask(self.current_df, project_id)
+                
+                reference_embedding_lst = self._process_embeddings(reference_df["embeddings"])
+                current_embedding_lst = self._process_embeddings(current_df["embeddings"])
+
+                reference_df = pd.DataFrame(reference_embedding_lst).sample(n=n_samples, random_state=random_state)
+                current_df = pd.DataFrame(current_embedding_lst).sample(n=n_samples, random_state=random_state)
+
+                if len(reference_df) and len(current_df):
```
Comment (k9845) [inline monitoring_scripts/featuredrift.py]: Shouldn't we check for length?

File: monitoring_scripts/featuredrift.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+if __name__ == "__main__":
+    reference_data_path = "csvfiles/data_with_embeddings_22000.csv"
+    current_data_path = "csvfiles/sampled_data_with_embeddings_testset.csv"
+    
+    reference_data_df = pd.read_csv(reference_data_path)
+    current_data_df = pd.read_csv(current_data_path)
+
+    feature_drift = FeatureDrift(reference_data_df, current_data_df)
+    df = feature_drift.compute_feature_drift(n_samples=10)
+    print(df)
```
Comment (k9845) [inline monitoring_scripts/featuredrift.py]:
```suggestion
```

File: monitoring_scripts/featuredrift.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                    temp_result["generated_at"] = datetime.date.today()
+
+                    final_result.append(temp_result)
+        
+        return pd.DataFrame.from_records(final_result)
+    
+
+if __name__ == "__main__":
+    reference_data_path = "csvfiles/data_with_embeddings_22000.csv"
+    current_data_path = "csvfiles/sampled_data_with_embeddings_testset.csv"
```
Comment (k9845) [inline monitoring_scripts/featuredrift.py]: This file path should be passed differently

File: monitoring_scripts/generate_outputs.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+from typing import List
+from tqdm import tqdm
+from botocore.exceptions import ClientError
+
+from postprocess_cpu_model_outputs import convert_current_dict_to_previous_one, get_predictions_all
+
+ENDPOINT_NAME = "main-model-cpu"
+AWS_REGION = "us-east-1"
+BATCH = 10
```
Comment (k9845) [inline monitoring_scripts/generate_outputs.py]: Let's pass this from `.env`

File: monitoring_scripts/generate_outputs.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    dataset_path = "csvfiles/test_v0.7.1.csv"
+    df = pd.read_csv(dataset_path).sample(n=10, random_state=1234).reset_index()
+
+    embeddings = ClassificationModelOutput(
+        df,
+        prediction_required=True,
+        embeddings_required=True
+    )
+    df = embeddings.generate_outputs()
+    #print(df)
```
Comment (k9845) [inline monitoring_scripts/generate_outputs.py]:
```suggestion
```

**Review by k9845 — CHANGES_REQUESTED** [review]

File: monitoring_scripts/generate_outputs.py (hunk 6 above)
Comment (ranjan-stha) [reply monitoring_scripts/generate_outputs.py]: reads from env

File: monitoring_scripts/featuredrift.py (hunk 3 above)
Comment (ranjan-stha) [reply monitoring_scripts/featuredrift.py]: fixed.

File: monitoring_scripts/featuredrift.py (hunk 2 above)
Comment (ranjan-stha) [reply monitoring_scripts/featuredrift.py]: fixed

File: monitoring_scripts/featuredrift.py (hunk 1 above)
Comment (ranjan-stha) [reply monitoring_scripts/featuredrift.py]: done

File: monitoring_scripts/featuredrift.py (hunk 5 above)
Comment (ranjan-stha) [reply monitoring_scripts/featuredrift.py]: just used for testing.

**Pull request closed without merging**